lib/combinePrimitives.js

```diff
@@ -0,0 +1,270 @@
+            }
+        }
+    };
+    var newBufferView = {
+        "buffer": createNewId(gltf, meshId, attributeKey, 'buffer'),
+        "byteOffset": 0,
+        "extras": {
+            "_pipeline": {
+                "deleteExtras": true
+            }
+        }
+    };
+    var newAccessor = {
+        "bufferView": createNewId(gltf, meshId, attributeKey, 'bufferView'),
+        "byteOffset": 0,
+        "byteStride": defaultValue(referenceAccessor.byteStride, 0),
+        "componentType": referenceAccessor.componentType,
+        "type": referenceAccessor.type,
+        "extras": {
+            "_pipeline": {
+                "deleteExtras": true
+            }
+        }
+    };
+
+    //Generate combined source for all grouped primitives.
+    for (var i = 0; i < primitiveGroup.length; i++) {
+        var primitive = primitiveGroup[i];
+        var accessor = defined(attributeType) ? accessors[primitive.attributes[attributeType]] : accessors[primitive.indices];
+
+        //If the current accessor has a different type or componentType, throw an error.
+        if (accessor.componentType !== componentType || accessor.type !== type) {
+            throw new DeveloperError('Attributes cannot reference accessors with different types or componentTypes.');
+        }
+
+        //Update the properties for the new accessor.
+        count += accessor.count;
+        if (defined(accessor.max)) {
+            if (defined(max)) {
+                for (var j = 0; j < max.length; j++) {
+                    if (accessor.max[j] > max[j]) {
+                        max[j] = accessor.max[j];
+                    }
+                }
+            }
+            else {
+                max = accessor.max;
+            }
+        }
+        if (defined(accessor.min)) {
+            if (defined(min)) {
+                for (var k = 0; k < min.length; k++) {
+                    if (accessor.min[k] < min[k]) {
+                        min[k] = accessor.min[k];
+                    }
+                }
+            }
+            else {
+                min = accessor.min;
+            }
+        }
+
+        //Obtain the source from the accessor's bufferview and append to the new source.
+        var bufferView = bufferViews[accessor.bufferView];
+        var viewByteOffset = bufferView.byteOffset;
+        var viewByteLength = defaultValue(bufferView.byteLength, 0);
+        var bufferSource = buffers[bufferView.buffer].extras._pipeline.source;
+        var bufferViewSource = bufferSource.slice(viewByteOffset, viewByteOffset + viewByteLength);
+
+        //Calculate the length of the accessor source using the count and byteStride or componentType.
+        var accessorByteLength = accessor.count;
+        if (accessor.byteStride > 0) {
+            accessorByteLength *= accessor.byteStride;
+        }
+        else {
+            //Int16
+            if (accessor.componentType === 5122 || accessor.componentType === 5123) {

```

Comment:
It is technically a private API in Cesium, but I suggest using https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Renderer/WebGLConstants.js here.